corehq/apps/linked_domain/remote_accessors.py

```diff
@@ -121,13 +121,18 @@ def fetch_remote_media(local_domain, missing_media, remote_app_details):
         media_class = CommCareMultimedia.get_doc_class(item['media_type'])
         content = _fetch_remote_media_content(item, remote_app_details)
         media_item = media_class.get_by_data(content)
-        media_item._id = item['multimedia_id']
+        if getattr(media_item, '_id', None) and not getattr(item, 'upstream_media_id', None):
+            # media already exists locally; update the multimedia_map
+            item['multimedia_id'], item['upstream_media_id'] = media_item._id, item['multimedia_id']
+        else:
+            media_item._id = item['multimedia_id']
         media_item.attach_data(content, original_filename=filename)
```

Comment:
For media that already exists locally, won't this create a duplicate `AuxMedia`?

Reply:
Yes, but only once (or on repeated force pulls of multimedia). So for example, a CommCareImage on my local testing app now has two `AuxMedia` entries that are identical except for:
```
"uploaded_filename": "aladdin-sane.jpeg"
"uploaded_filename": "module0_case_search_again_label_item_en.jpeg"
```

Because only the right media is now showing up in `missing_multimedia`, it doesn't get fetched/attached again after the first time. This seems comparable to the behavior I see when uploading the same file to multiple apps locally - a new `AuxMedia` attachment is created once, but in the future the app knows it already exists and doesn't try to add it again.